src/screens/Profiles/UserLibrary/components/UserLibraryEditScreen/component.js

```diff
@@ -145,169 +154,203 @@ export class UserLibraryEditScreenComponent extends React.Component {
       }
     }
 
-    await updateUserLibraryEntry(libraryType, libraryStatus, {
-      id: libraryEntry.id,
-      finishedAt,
-      notes: (notes && notes.trim()),
-      progress,
-      ratingTwenty,
-      reconsumeCount,
-      startedAt,
-      status,
-      private: isPrivate,
-    });
+    this.setState({ loading: true, error: null });
 
-    this.props.navigation.goBack();
+    try {
+      await updateUserLibraryEntry(libraryType, libraryStatus, {
+        id: libraryEntry.id,
+        finishedAt,
+        notes: (notes && notes.trim()),
+        progress,
+        ratingTwenty,
+        reconsumeCount,
+        startedAt,
+        status,
+        private: isPrivate,
+      });
+      this.setState({ loading: false });
+      this.props.navigation.goBack();
+    } catch (e) {
+      this.setState({ loading: false, error: e });
+      console.log(e);
+    }
   }
 
   render() {
     const { canEdit, libraryEntry, libraryType, ratingSystem } = this.props.navigation.state.params;
+    const { loading, error } = this.state;
+
+    const rightTitle = loading ? 'Saving' : 'Save';
+    const leftTitle = loading ? null : 'Cancel';
+
     const maxProgress = this.getMaxProgress();
+
     // { value: 'current', anime: 'Currently Watching', manga: 'Currently Reading' },
-    const status = STATUS_SELECT_OPTIONS.filter(item => item.value === this.state.status)[0][libraryType]
+    const status = STATUS_SELECT_OPTIONS.filter(item => item.value === this.state.status)[0][libraryType];
+
     return (
       <View style={styles.container}>
         <SimpleHeader
-          leftAction={this.props.navigation.goBack}
-          leftContent={canEdit ? 'Cancel' : 'Done'}
+          leftAction={!loading && this.props.navigation.goBack}
+          leftContent={canEdit ? leftTitle : 'Done'}
           titleContent={canEdit ? 'Edit Entry' : 'Details'}
-          rightContent={canEdit ? 'Save' : null}
-          rightAction={this.saveEntry}
+          rightContent={canEdit ? rightTitle : null}
+          rightAction={!loading && this.saveEntry}
         />
+        {/* Container for the content below the header */}
+        <View style={{ flex: 1 }}>
 
-        <KeyboardAwareScrollView>
-          {/* Status */}
-          <View style={styles.editRow}>
-            <Text style={styles.editRowLabel}>
-              Library Status:
-            </Text>
-            <SelectMenu
-              disabled={!canEdit}
-              options={this.selectOptions}
-              onOptionSelected={this.onStatusChanged}
-            >
-              <Text style={styles.editRowValue}>
-                {status}
+          {/* Loading */}
+          {loading &&
+            <View style={styles.loadingContainer}>
+              <ActivityIndicator size="large" color="white" />
+            </View>
+          }
+
+          {/* Error */}
+          {error &&
```

Comment:
if `error` is `''` then we'll have an of strings should be placed inside <Text> etc. Let's change it to `isNull(error)` from `lodash`; or something similar